core/01_core.js

```diff
@@ -235,7 +242,7 @@
   }
 
   function opAsync(opName, ...args) {
-    return ops[opName](...args);
+    return ops[opName](...new SafeArrayIterator(args));
```

Comment:
Thanks for the patch but this likely has negative performance implications.

can you check the before and after of op_boop bench?

Reply:
Using `ReflectApply` seemed to perform better than using `SafeArrayIterator`.

main branch

```
running 3 tests
test bench_op_async   ... bench:     349,345 ns/iter (+/- 7,851)
test bench_op_nop     ... bench:      37,275 ns/iter (+/- 2,333)
test bench_op_pi_json ... bench:      41,590 ns/iter (+/- 2,003)
```

use `SafeArrayIterator`

```
running 3 tests
test bench_op_async   ... bench:     460,511 ns/iter (+/- 8,330)
test bench_op_nop     ... bench:      36,833 ns/iter (+/- 2,021)
test bench_op_pi_json ... bench:      41,529 ns/iter (+/- 2,572)
```

use `ReflectApply`

```
running 3 tests
test bench_op_async   ... bench:     348,736 ns/iter (+/- 11,100)
test bench_op_nop     ... bench:      37,114 ns/iter (+/- 2,677)
test bench_op_pi_json ... bench:      41,799 ns/iter (+/- 2,537)
```